contrib/IECoreUSD/src/IECoreUSD/USDScene.cpp

```diff
@@ -0,0 +1,832 @@
+}
+
+Imath::V3d convert( const pxr::GfVec3d &src )
+{
+	return Imath::V3d( src[0], src[1], src[2] );
+}
+
+Imath::V3f convert( const pxr::GfVec3f &src )
+{
+	return Imath::V3f( src[0], src[1], src[2] );
+}
+
+Imath::V2d convert( const pxr::GfVec2d &src )
+{
+	return Imath::V2d( src[0], src[1] );
+}
+
+Imath::V2f convert( const pxr::GfVec2f &src )
+{
+	return Imath::V2f( src[0], src[1] );
+}
+
+Imath::Box3d convert( const pxr::GfBBox3d &src )
+{
+	const auto &srcBox = src.GetBox();
+
+	return Imath::Box3d( convert( srcBox.GetMin() ), convert( srcBox.GetMax() ) );
+}
+
+Imath::M44d convert( const pxr::GfMatrix4d &src )
+{
+	Imath::M44d r;
+	for( int i = 0; i < 4; ++i )
+	{
+		for( int j = 0; j < 4; ++j )
+		{
+			r[i][j] = src[i][j];
+		}
+	}
+
+	return r;
+}
+
+IECore::IntVectorDataPtr convert( const pxr::VtIntArray &data )
+{
+	IECore::IntVectorDataPtr newData = new IECore::IntVectorData();
+
+	std::vector<int> &writable = newData->writable();
+	writable.resize( data.size() );
+	for( size_t i = 0; i < data.size(); ++i )
+	{
+		writable[i] = data[i];
+	}
+	return newData;
+}
+
+IECore::V3fVectorDataPtr convert( const pxr::VtVec3fArray &data )
+{
+	IECore::V3fVectorDataPtr newData = new IECore::V3fVectorData();
+
+	std::vector<Imath::V3f> &writable = newData->writable();
+	writable.resize( data.size() );
+	for( size_t i = 0; i < data.size(); ++i )
+	{
+		writable[i] = Imath::V3f( data[i][0], data[i][1], data[i][2] );
+	}
+	return newData;
+}
+
+IECore::PrimitiveVariable::Interpolation convertInterpolation( pxr::TfToken interpolationToken )
+{
+	if( interpolationToken == pxr::UsdGeomTokens->varying )
+	{
+		return IECore::PrimitiveVariable::Varying;
+	}
+	if( interpolationToken == pxr::UsdGeomTokens->vertex )
+	{
+		return IECore::PrimitiveVariable::Vertex;
+	}
+	if( interpolationToken == pxr::UsdGeomTokens->uniform )
+	{
+		return IECore::PrimitiveVariable::Uniform;
+	}
+	if( interpolationToken == pxr::UsdGeomTokens->faceVarying )
+	{
+		return IECore::PrimitiveVariable::FaceVarying;
+	}
+	if( interpolationToken == pxr::UsdGeomTokens->constant )
+	{
+		return IECore::PrimitiveVariable::Constant;
+	}
+
+	return IECore::PrimitiveVariable::Invalid;
+}
+
+template<typename DestElementType, typename SourceElementType>
+typename IECore::GeometricTypedData<std::vector<DestElementType> >::Ptr convertIt( const pxr::VtValue &value )
+{
+	typedef pxr::VtArray<SourceElementType> SourceArrayType;
+	typedef IECore::GeometricTypedData<std::vector<DestElementType> > DestArrayType;
+	typedef std::vector<DestElementType> DestStorageType;
+
+	if( value.IsHolding<SourceArrayType>() )
+	{
+		const auto &r = value.Get<SourceArrayType>();
+
+		typename DestArrayType::Ptr d = new DestArrayType();
+		DestStorageType &t = d->writable();
+
+		t.resize( r.size() );
+
+		for( size_t i = 0; i < r.size(); ++i )
+		{
+			t[i] = convert( r[i] );
+		}
+
+		return d;
+	}
+
+	return nullptr;
+}
+
+IECore::PrimitiveVariable fixForNow( const IECore::PrimitiveVariable &srcPrimVar )
```

Comment:
This should be removed once #631 has been merged.

Reply:
Just the GL part.

Reply:
May I propose that we just merge that single commit then, to give me time to consider the iterator API properly?

Reply:
yup.